Python/Product/Debugger/Debugger/DebugConnectionListener.cs

```diff
@@ -95,39 +98,64 @@ static class DebugConnectionListener {
             var stream = new NetworkStream(socket, ownsSocket: true);
             try {
                 socket.Blocking = true;
-                string debugId = stream.ReadString();
-                var result = (ConnErrorMessages)stream.ReadInt32();
 
-                lock (_targets) {
-                    Guid debugGuid;
-                    WeakReference weakProcess;
-                    PythonProcess targetProcess;
-
-                    if (Guid.TryParse(debugId, out debugGuid) &&
-                        _targets.TryGetValue(debugGuid, out weakProcess) &&
-                        (targetProcess = weakProcess.Target as PythonProcess) != null) {
-
-                        if (result == ConnErrorMessages.None) {
-                            targetProcess.Connect(stream);
-                            stream = null;
-                            socket = null;
+                var debugConn = new DebugConnection(stream);
+                try {
+                    string debugId = string.Empty;
+                    var result = ConnErrorMessages.None;
+                    using (var connectedEvent = new AutoResetEvent(false)) {
+                        EventHandler<LDP.LocalConnectedEvent> eventReceived = (object sender, LDP.LocalConnectedEvent ea) => {
```

Comment:
Agreed, mainly for the benefit of those people who are building PTVS themselves, to save them having to do too much work to get it to build.